Packs/Syslog/Integrations/SyslogSender/SyslogSender.py

```diff
@@ -54,60 +61,135 @@
+        ssl_sock = ssl_context.wrap_socket(ssl_sock, server_hostname=self.address)
+        self.socket = ssl_sock
+        try:
+            self.socket.connect((self.address, self.port))
+        except OSError as exc:
+            if ssl_sock:
+                ssl_sock.close()
+            raise DemistoException(str(exc))
+
+    def emit(self, record):
+        """
+        Emit a record.
+
+        The record is formatted, and then sent to the syslog server. If
+        exception information is present, it is NOT sent to the server.
+        """
+        ident = ''          # prepended to all messages
+        try:
+            msg = self.format(record)
+            if ident:
+                msg = ident + msg
+
+            # Calculate the priority value
+            priority = (self.facility << 3) | self.level
+            # Construct the syslog message in RFC 5424 format
+            syslog_message = '<{priority}>1 {timestamp} {hostname} {appname} {procid} {msgid} - {message}\n'.format(
+                priority=priority,
+                timestamp=datetime.utcnow().strftime('%Y-%m-%dT%H:%M:%S.%fZ'),
+                hostname=socket.gethostname(),
+                appname=record.name,
+                procid=os.getpid(),
+                msgid='-',
+                message=self.format(record)
+            )
+            # Connect to the syslog server
+            self.socket.send(syslog_message.encode('utf-8'))
+
+        except Exception as e:
+            if self.socket:
+                self.socket.close()
+            demisto.error(str(e))
+
 
 ''' Syslog Manager '''
 
 
 class SyslogManager:
-    def __init__(self, address: str, port: int, protocol: str, logging_level: int, facility: int):
+    def __init__(self, address: str, port: int, protocol: str, logging_level: int,
+                 facility: int, cert_path: str | None, self_signed_certificate: bool):
         """
         Class for managing instances of a syslog logger.
         :param address: The IP address of the syslog server.
         :param port: The port of the syslog server.
-        :param protocol: The messaging protocol (TCP / UDP).
+        :param protocol: The messaging protocol (TCP / UDP / TLS).
         :param logging_level: The logging level.
         """
         self.address = address
         self.port = port
         self.protocol = protocol
         self.logging_level = logging_level
         self.facility = facility
+        self.syslog_cert_path = cert_path
+        self.self_signed_cert = self_signed_certificate
 
     @contextmanager  # type: ignore[misc, arg-type]
     def get_logger(self) -> Generator:
         """
         Get a new instance of a syslog logger.
         :return: syslog logger
         """
-        handler = self._get_handler()
+        if self.protocol == TLS and self.syslog_cert_path:
+            handler = self.init_handler_tls(self.syslog_cert_path)
+            demisto.debug('get handler tls')
```

Comment:
Can this init fail? if so it's a better idea to print the log before it.
One more thing - I don't understand what "get handler tls" means. Perhaps `creating tls logger handler`